lighthouse-core/audits/legacy-javascript.js

```diff
@@ -0,0 +1,334 @@
+/**
+ * @license Copyright 2019 Google Inc. All Rights Reserved.
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except in compliance with the License. You may obtain a copy of the License at http://www.apache.org/licenses/LICENSE-2.0
+ * Unless required by applicable law or agreed to in writing, software distributed under the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the License for the specific language governing permissions and limitations under the License.
+ */
+'use strict';
+
+/** @typedef {{name: string, expression: string}} Pattern */
+/** @typedef {{name: string, line: number, column: number}} PatternMatchResult */
+
+const Audit = require('./audit.js');
+const NetworkRecords = require('../computed/network-records.js');
+const i18n = require('../lib/i18n/i18n.js');
+
+const UIStrings = {
+  /** Title of a Lighthouse audit that tells the user about legacy polyfills and transforms used on the page. This is displayed in a list of audit titles that Lighthouse generates. */
+  title: 'Legacy JavaScript',
+  // eslint-disable-next-line max-len
+  description: 'Polyfills and transforms enable legacy browsers to use new JavaScript features. However, many aren\'t necessary for modern browsers. For your bundled JavaScript, adopt a modern script deployment strategy using module/nomodule feature detection to reduce the amount of code shipped to modern browsers, while retaining support for legacy browsers. [Learn More](https://philipwalton.com/articles/deploying-es2015-code-in-production-today/)',
```

Comment:
Split this into ''+'' for line length, lets keep it below 100. Is there a web.dev link for this maybe?  @egsweeny

Reply:
but why. word wrap is a thing

Reply:
also +1 to the web.dev thing. there is @housseindjirdeh guide on web.dev which I also want to link here but I forgot how to find it